errors.go

```diff
@@ -7,6 +7,43 @@ import (
 	"github.com/pkg/errors"
 )
 
+type JobNotFoundError struct {
+	msg string
+}
+
+func (e *JobNotFoundError) Error() string { return e.msg }
+
+// NewJobNotFoundError creates a new error indicating that a job could not be
+// found in the queue.
+func NewJobNotFoundError(msg string) error { return &JobNotFoundError{msg: msg} }
+
+// NewJobNotFoundErrorf creates a new error with a formatted message, indicating
+// that a job could not be found in the queue.
+func NewJobNotFoundErrorf(msg string, args ...interface{}) error {
+	return NewJobNotFoundError(fmt.Sprintf(msg, args...))
+}
+
+// MakeJobNotFoundError constructs an error from an existing one, indicating
+// that a job could not be found in the queue.
+func MakeJobNotFoundError(err error) error {
+	if err == nil {
+		return nil
+	}
+
+	return NewJobNotFoundError(err.Error())
+}
+
+// IsJobNotFound tests an error it is due tonot being able to find a job in the
```

Comment:
I would change this to something like: `IsJobNotFound checks if the error is a job not found error.`

Reply:
I tweaked it a bit, but I think it's not that useful if all it does is restate the function's name.